### PR TITLE
Prevent using both `authorizeRequests` and `authorizeHttpRequests`

### PR DESCRIPTION
**Pull request by marcusdacoregio**
Closes gh-10573

### CODE REVIEW

**Comment by marcusdacoregio** [comment]
Thanks for the feedback @rwinch.

Totally agreed. Users who add filters manually know exactly what they are doing. I'm going to apply the changes.

**Comment by marcusdacoregio** [comment]
I've updated the PR to avoid having `authorizeHttpRequests` and `authorizeRequests` in the same configuration.

File: config/src/main/java/org/springframework/security/config/annotation/web/configurers/AuthorizeHttpRequestsConfigurer.java [hunk 1]
```diff
@@ -70,8 +70,13 @@ public AuthorizationManagerRequestMatcherRegistry getRegistry() {
 		return this.registry;
 	}
 
+	@SuppressWarnings("unchecked")
 	@Override
 	public void configure(H http) {
+		ExpressionUrlAuthorizationConfigurer<?> configurer = http
```
Comment (rwinch) [inline config/src/main/java/org/springframework/security/config/annotation/web/configurers/AuthorizeHttpRequestsConfigurer.java:76]: Can we move this to where we are building http? I'd prefer the configurers not need to know about each other.

**Review by rwinch — CHANGES_REQUESTED** [review]

**Comment by marcusdacoregio** [comment]
Thanks @rwinch. I've updated the PR.

**Review by rwinch — APPROVED** [review]
The code looks great!

The only other change is the PR should be pointed to 5.7 branch rather than main since the ticket is (rightfully) assigned to 5.7

**Pull request merged**